analyzers/tests/SonarAnalyzer.UnitTest/Extensions/EnumerableExtensionsTest.cs

```diff
@@ -173,6 +176,80 @@ public StructType(int count)
                 this.count = count;
             }
         }
+
+        [TestMethod]
+        public void JoinAndNull()
+        {
+            var result = ((object[])null).JoinAnd();
+            result.Should().Be(string.Empty);
+        }
+
+        [DataTestMethod]
+        [DataRow("")]
+        [DataRow("", null)]
+        [DataRow("", "")]
+        [DataRow("", "", "")]
+        [DataRow("", "", " ", "\t", null)]
+        [DataRow("a", "a")]
+        [DataRow("a and b", "a", "b")]
+        [DataRow("a and b", "a", null, "b")]
+        [DataRow("a, b, and c", "a", "b", "c")]
+        [DataRow("a, b, and c", "a", null, "b", "", "c")]
+        public void JoinAndStrings(string expected, params string[] collection)
+        {
+            var result = collection.JoinAnd();
+            result.Should().Be(expected);
+        }
+
+        [DataTestMethod]
+        [DataRow("")]
+        [DataRow("0", 0)]
+        [DataRow("0 and 1", 0, 1)]
+        [DataRow("0, 1, and 2", 0, 1, 2)]
+        [DataRow("1000", 1000)]
+        public void JoinAndInts(string expected, params int[] collection)
+        {
+            var result = collection.JoinAnd();
+            result.Should().Be(expected);
+        }
+
+        [DataTestMethod]
+        [DataRow("")]
+        [DataRow("08/30/2022 12:29:11", "2022-08-30T12:29:11")]
+        [DataRow("08/30/2022 12:29:11 and 12/24/2022 16:00:00", "2022-08-30T12:29:11", "2022-12-24T16:00:00")]
+        [DataRow("08/30/2022 12:29:11, 12/24/2022 16:00:00, and 12/31/2022 00:00:00", "2022-08-30T12:29:11", "2022-12-24T16:00:00", "2022-12-31T00:00:00")]
+        public void JoinAndDateTime(string expected, params string[] collection)
+        {
+            var oldLocale = Thread.CurrentThread.CurrentCulture;
+            var oldUiLocale = Thread.CurrentThread.CurrentUICulture;
+            try
+            {
```

Comment:
This `save/try/finally/restore` is also used in `DisablingRequestValidation_CS_WebConfig_SubFolders`. What about extracting it to `IDisposable` object similar to `EnvironmentVariableScope`? Some kind of
```
using var scope = new CurrentCultureScope(CultureInfo.InvariantCulture)`
```

Reply:
Another option I used in the past is to introduce a "CultureDataTestMethodAttribute". I placed a sample in the file. This could come in handy in other places too. If you like it I would move it to a better place.

Reply:
That will not work for the `DisablingRequestValidation_CS_WebConfig_SubFolders`, as this supports only constant cultures.

Let's use the scope. This one is too indirect.